log-analyzer/features/org.wso2.carbon.la.apim.feature/src/main/resources/gadgets/LogAPIMArtifactDeleted/js/VizGrammar.js

```diff
@@ -0,0 +1,2300 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var arc = function(dataTable, config) {
+    this.metadata = dataTable[0].metadata;
+    var marks =[];
+    this.spec = {};
+
+    config = checkConfig(config, this.metadata);
+    this.config = config;
+    dataTable[0].name= config.title;
+
+    dataTable[0].transform = [{
+        "type": "pie",
+        "field": this.metadata.names[config.x]
+    },
+        {
+            "type": "formula",
+            "field": "percentage",
+            "expr": "datum."+this.metadata.names[config.x]+" / 360 * 100"
+        }];
+
+    var scales =  [];
+
+    if (config.colorDomain == -1) {
+        config.colorDomain = {"data":  config.title, "field": this.metadata.names[config.color]};
+    }
+
+
+    var colorScale = {
+        "name": "color",
+        "type": "ordinal",
+        "domain": config.colorDomain,
+        "range": config.colorScale
+    };
+    scales.push(colorScale);
+    marks.push(getPieMark(config, this.metadata));
+
+    if (config.percentage) {
+        marks.push(getPieText(config, this.metadata));
+    }
+
+
+    var legendTitle = "Legend";
+
+    if (config.title != "table") {
+        legendTitle = config.title;
+    }
+
+    var legends = [
+        {
+            "fill": "color",
+            "title": "Legend",
+            "offset": 20,
+            "properties": {
+                "symbols": {
+                    "fillOpacity": {"value": 0.5},
+                    "stroke": {"value": "transparent"}
+                }
+            }
+        }
+    ];
+
+    this.spec.legends = legends;
+    this.spec.width = config.width;
+    this.spec.height = config.height;
+    this.spec.data = dataTable;
+    this.spec.scales = scales;
+    this.spec.padding = config.padding;
+    this.spec.marks = marks;
+};
+
+arc.prototype.draw = function(div, callbacks) {
+
+    var viewUpdateFunction = (function(chart) {
+        if(this.config.tooltip.enabled){
+            this.config.tooltip.type = "arc";
+            createTooltip(div);
+            this.view = chart({el:div}).renderer(this.config.renderer).update();
+            bindTooltip(div,this.view,this.config,this.metadata);
+        } else {
+            this.view = chart({el:div}).renderer(this.config.renderer).update();
+        }
+
+        if (callbacks != null) {
+            for (var i = 0; i<callbacks.length; i++) {
+                this.view.on(callbacks[i].type, callbacks[i].callback);
+            }
+        }
+
+    }).bind(this);
+
+    if(this.config.maxLength != -1){
+        var dataset = this.spec.data[0].values;
+        var maxValue = this.config.maxLength;
+        if(dataset.length >= this.config.maxLength){
```

Comment:
Hi Chandana,
We have replace the Vizgrammer.js with Vizgrammer.min.js and relevant commits are in

https://github.com/wso2/shared-analytics/pull/126/files#diff-6ac780d61b2b6d470c8ff04bdeed3bcdR52

Thanks,
SajithD